### ISSUE TITLE
Run time

### ISSUE THREAD

**Issue by yetian28** [TDeVries/enas_pytorch#2]
Hi, 
Thanks for the code sharing!

I tried to read it and run it.  10 epochs in the main function take one Nividia Tesla P100 12G 15h to run. Am I wrong or it just takes this amount of time?

**Comment by TDeVries** [comment]
Hi, I ran the code on my local machine (Titan X Pascal) and I'm getting about 5 or 6 minutes per epoch. 15 hours for 10 definitely seems excessive.

**Comment by yetian28** [comment]
Oh, I see. Thank you for the reply! I must be doing something stupid.

I am using WINDOWS Anaconda cuda9. I hope this is not the reason?
I commented out visdom part. Should be fine.
I also comment out parts related to [torch.save(state, filename)]. I guess this part is saving the model purely because I don't see function calling the saved model.  Is this the case?

**Comment by yetian28** [comment]
Yes when it comes to controller training it's supper slow.

**Comment by TDeVries** [comment]
Yes, I'm pretty sure this is because it does multiple gradient accumulations in order to get a cleaner gradient signal (default is 20 iterations per update). You can change how many accumulations it does by changing the controller_num_aggregate argument. Reducing that value should speed up training, but you may also experience some instability.

**Comment by yetian28** [comment]
Thanks! I changed it to 2 and get 1 epoch for 20 minute. Maybe I should try 1...
Anyway, thanks so much.

**Comment by TDeVries** [comment]
Are you sure the GPU is being used? This sounds kind of like the speeds that would be expected if trained only on CPU.

**Comment by yetian28** [comment]
I uus GPU-Z to check it just now. It seems that it is working.
![Capture](https://user-images.githubusercontent.com/35240846/61090471-a446e200-a3f3-11e9-9401-3509f17fe92d.PNG)
